lib/rdoc/generator/template/darkfish/js/darkfish.js

```diff
@@ -106,9 +106,22 @@ function hookSidebar() {
   }
 }
 
+function hookAnchorLinks() {
+  document.addEventListener('click', (e) => {
+    const anchor = e.target.closest('#navigation a[href^="#"]');
```

Comment:
We can't restrict the target to just anchors as some of the links (e.g. in search results) have relative path:
<img width="504" alt="Screenshot 2024-12-17 at 19 34 04" src="https://github.com/user-attachments/assets/dbe02083-8b4c-45f0-bd67-cc5547dac6f8" />

Reply:
I think that would still cause a refresh of the page, making listening to those click events unnecessary (switching a page hides the navigation anyways on mobile by default)